.github/workflows/nix.yaml

```diff
@@ -0,0 +1,35 @@
+name: nix-build
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-push-oci:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ vars.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+    - uses: cachix/install-nix-action@v25
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
```

Comment:
any reason we use unstable here instead of 23.11?

Reply:
The reason to use unstable in CI is to catch early warnings of breakages that might occur for users of unstable, and to rely on packages that are not yet available in a stable branch. However, I think it's actually preferable to use 23.11 as the version we recommend to minimize support issues and only migrate to unstable if necessary to pull in new packages, and we can use overlays to add packages we specifically want to depend on in many cases.